events/handlers.js

```diff
@@ -0,0 +1,184 @@
+    let results = [];
+    fs.createReadStream(filePath)
+      .pipe(csv())
+      .on("data", (data) => {
+        if (
+          req.query.location === undefined ||
+          req.query.location == data.location
+        ) {
+          results.push(data);
+        }
+      })
+      .on("end", () => {
+        res.send(results);
+      });
+  });
+}
+
+function createEvent(req, res) {
+  lockEvents(req, res, (req, res) => {
+    // just add at the end of the file
+    fs.open(filePath, "a", 666, function (e, id) {
+      // convert JSON array to CSV string
+      converter.json2csv(
+        req.body,
+        (err, csv) => {
+          if (err) {
+            res.status(500).send();
+            return;
+          }
+          const row = "\n" + csv;
+          fs.write(id, row, null, "utf8", function () {
+            fs.close(id, function () {
+              res.status(200).send();
+            });
+          });
+        },
+        { prependHeader: false }
+      );
+    });
+  });
+}
+
+function updateEvent(req, res) {
+  lockEvents(req, res, (req, res) => {
+    let results = [];
+    let found = false;
+    fs.createReadStream(filePath)
+      .pipe(csv())
+      .on("data", (data) => {
+        if (data.id == req.params.eventId) {
+          found = true;
+          results.push(req.body);
+        } else {
+          results.push(data);
+        }
+      })
+      .on("end", () => {
+        if (found === false) {
+          res.status(404).send("event not found");
+          return;
+        }
+        const csvWriter = createCsvWriter({
+          path: filePath,
+          header: headerFile,
+        });
+        csvWriter.writeRecords(results).then(() => {
+          res.status(200).send();
+        });
+      });
+  });
+}
+
+function deleteEvent(req, res) {
+  lockEvents(req, res, (req, res) => {
+    let results = [];
+    let found = false;
+    fs.createReadStream(filePath)
+      .pipe(csv())
+      .on("data", (data) => {
+        if (data.id == req.params.eventId) {
+          found = true;
+          return;
+        }
+        results.push(data);
+      })
+      .on("end", () => {
+        if (found === false) {
+          res.status(404).send("event not found");
+          return;
+        }
+        const csvWriter = createCsvWriter({
+          path: filePath,
+          header: headerFile,
+        });
+        csvWriter.writeRecords(results).then(() => {
+          console.log("The CSV file was written successfully");
+          res.status(200).send();
+        });
+      });
+  });
+}
+
+function getEventsBatch(req, res) {
+  lockEvents(req, res, (req, res) => {
+    let results = [];
+    fs.createReadStream(filePath)
+      .pipe(csv())
+      .on("data", (data) => {
+        results.push(data);
+      })
+      .on("end", () => {
+        res.send(results);
+      });
+  });
+}
+
+function lockEvents(req, res, callback) {
+  lockFile.lock("some-file.lock", {}, function (er) {
```

Comment:
We might need to set some options to `lock` function instead of `{}`. https://github.com/npm/lockfile#optsretrywait
Maybe default options would work for us, but from the quick view at implementation it doesn't look like that https://github.com/npm/lockfile/blob/master/lockfile.js#L137

In case of incorrect options we just may not retry to lock file when it is already locked.